src/containers/ImageUploader/components/ImageUploadFormElement.tsx

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) 2025-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+import { useFormikContext } from "formik";
+import { useTranslation } from "react-i18next";
+import { DeleteBinLine, UploadCloudLine } from "@ndla/icons";
+import { ImageMeta } from "@ndla/image-search";
+import {
+  Button,
+  FieldRoot,
+  FileUploadDropzone,
+  FileUploadHiddenInput,
+  FileUploadLabel,
+  FileUploadRoot,
+  FileUploadTrigger,
+  FieldErrorMessage,
+  IconButton,
+} from "@ndla/primitives";
+import { SafeLink } from "@ndla/safelink";
+import { styled } from "@ndla/styled-system/jsx";
+import { FormField } from "../../../components/FormField";
+import { MAX_IMAGE_UPLOAD_SIZE } from "../../../constants";
+import { ImageFormikType } from "../imageTransformers";
+
+const StyledImg = styled("img", {
+  base: {
+    borderRadius: "xsmall",
+  },
+});
+
+const StyledIconButton = styled(IconButton, {
+  base: {
+    position: "absolute",
+    right: "-large",
+  },
+});
+
+const ImageContentWrapper = styled("div", {
+  base: {
+    position: "relative",
+  },
+});
+
+interface Props {
+  language: string;
+}
+
+export const ImageUploadFormElement = ({ language }: Props) => {
+  const { t } = useTranslation();
+  const formikContext = useFormikContext<ImageFormikType>();
+  const { values, setFieldValue } = formikContext;
+
+  // We use the timestamp to avoid caching of the `imageFile` url in the browser
+  const timestamp = new Date().getTime();
+  const imgSrc = values.filepath || `${values.imageFile}?width=800&ts=${timestamp}`;
+
+  return (
+    <ImageContentWrapper>
+      {!values.imageFile && (
+        <FormField name="imageFile">
+          {({ helpers, meta }) => (
+            <FieldRoot required invalid={!!meta.error}>
+              <FileUploadRoot
+                accept={["image/gif", "image/png", "image/jpeg", "image/jpg", "image/svg+xml"]}
+                onFileAccept={(details) => {
+                  const file = details.files?.[0];
+                  if (!file) return;
+                  setFieldValue("filepath", URL.createObjectURL(file));
+                  Promise.resolve(
+                    createImageBitmap(file as Blob).then((image) => {
+                      setFieldValue("imageDimensions", image);
+                    }),
+                  );
+                  setFieldValue("imageFile", file);
+                  setFieldValue("contentType", file.type);
+                  setFieldValue("fileSize", file.size);
```

Comment:
Hvordan funker dette med det nye du har laget? Setter du masse felter som egentlig ikke eksisterer i skjemaet? Er det problematisk mtp dirty?

Reply:
Hva tenker du på?
Dirty vil kanskje alltid bli satt om du har rørt formet siden slette knappen ikke sletter alle feltene den har satt?
Sånn dirtyness fungerer jo sånn vi har lyst til nå antar jeg iom at validering slår inn når man sletter et bilde.

Om jeg unsetter disse ved sletting av bildet så vil vi kanskje krølle det til andre plasser? Hva tycks?

Reply:
På sikt lurer jeg på om vi skal dra dette (FileUploadRoot) ut i sin egen komponent, og heller tvinge konsumenter til å sette de field-verdiene som trengs. Må dog ikke gjøres her, men kanskje legge inn en TODO?

Reply:
Har vi noen fremgangsmåte for hvordan vi dealer med TODO's?

Jeg syns det litt lett blir sånn at de bare ligger i kodebasen for alltid og ingen tar ansvar for de.

Reply:
Nei, egentlig ikke. Håpet er vel at noen tar tak i det når vi har ekstra tid :')